container/lxd/server.go

```diff
@@ -225,6 +228,30 @@ func addRemoveReplaceProfileName(profiles []string, oldProfile, newProfile strin
 	return profiles
 }
 
+// UpdateContainerProfiles applies the given profiles (by name) to the
+// named container.  It is assumed the profiles have all been added to
+// the server before hand.
+func (s *Server) UpdateContainerProfiles(name string, profiles []string) error {
+	container, eTag, err := s.GetContainer(name)
+	if err != nil {
+		return errors.Trace(errors.Annotatef(err, "failed to get %q", name))
+	}
+
+	container.Profiles = profiles
+	resp, err := s.UpdateContainer(name, container.Writable(), eTag)
+	if err != nil {
+		return errors.Trace(errors.Annotatef(err, "failed to update %q with profiles", name))
+	}
+
+	op := resp.Get()
+	logger.Debugf("updated %q, waiting on %s", name, op.Description)
+	err = resp.Wait()
+	if err != nil {
+		logger.Tracef("updating %q failed on %q", name, err)
+	}
+	return errors.Trace(err)
```

Comment:
I'm unsure we need the trace if we're already passing the error back

```suggestion
	if err := resp.Wait(); err != nil {
		logger.Tracef("updating %q failed on %q", name, err)
		return errors.Trace(err)
	}
	return nil
```

Reply:
removed